### ISSUE TITLE
Strapi components doesn't work for me

### ISSUE THREAD

**Issue by Carduelis** [erikvullings/strapi-to-typescript#24]
Hi! I see errors like this:
```
type 'page' unknown on IHomepage[page] => fallback to 'any'. Add in the input arguments the folder that contains *.settings.json with info.name === 'page'
```

My config file is:
```
input: [
    "api",
    "./node_modules/strapi-plugin-upload/models/",
    "./extensions/users-permissions/models/",
  ],
  inputGroup: "./components",
  output: "../next/typings/",
```

My `homepage.settings.json` is:
```
{
  "kind": "singleType",
  "collectionName": "homepages",
  "info": {
    "name": "Homepage",
    "description": ""
  },
  "options": {
    "increments": true,
    "timestamps": true,
    "draftAndPublish": true
  },
  "attributes": {
    "title": {
      "type": "string"
    },
    "description": {
      "type": "text"
    },
    "page": {
      "type": "component",
      "repeatable": false,
      "component": "meta.page"
    }
  }
}
```

My components folder is like:
```
/components
-- common/
---- page.json
---- meta-tag.json
---- open-graph.json
-- meta/
```

My `page.json` is like:

```
{
  "collectionName": "components_meta_pages",
  "info": {
    "name": "page",
    "icon": "file-powerpoint"
  },
  "options": {},
  "attributes": {
    "seo": {
      "type": "component",
      "repeatable": false,
      "component": "meta.seo"
    },
    "metaTags": {
      "type": "component",
      "repeatable": true,
      "component": "meta.meta-tag"
    },
    "openGraph": {
      "type": "component",
      "component": "meta.open-graph"
    }
  }
}
```

I checked [this](https://github.com/erikvullings/strapi-to-typescript/issues/9) and [this](https://github.com/erikvullings/strapi-to-typescript/issues/2) issues, didn't help.

I also checked source code, but wasn't able to figure out the solution.

**Comment by Carduelis** [comment]
But as far as I got this error and tried to figure out the reason, I have a suggestion for better debug errors:

Instead of
```
const findModelName = (n: string) => {
      const result = findModel(this.strapiModels, n);
      if (!result && n !== '*') console.debug(`type '${n}' unknown on ${interfaceName}[${name}] => fallback to 'any'. Add in the input arguments the folder that contains *.settings.json with info.name === '${n}'`)
      return result ? result.interfaceName : 'any';
    };
```

We can make this:
```
const createDebugMessage = (
  attribute: IStrapiModelAttribute,
  n: string,
  interfaceName: string,
  name: string
) => {
  if (attribute.component) {
    return `type '${n}' unknown on ${interfaceName}[${name}] => fallback to 'any'. Add in the inputGroup arguments the folder that contains a component`;
  }
  if (attribute.model && attribute.plugin) {
    return `type '${n}' unknown on ${interfaceName}[${name}] => fallback to 'any'. Add in the input arguments the folder that contains definitions of "${attribute.plugin}" plugin`;
  }

  return `type '${n}' unknown on ${interfaceName}[${name}] => fallback to 'any'. Add in the input arguments the folder that contains *.settings.json with info.name === '${n}'`;
};


const findModelName = (n: string) => {
      const result = findModel(this.strapiModels, n);
      if (!result && n !== '*') {
          // variable 'a' comes from strapiModelAttributeToProperty method
          const message = createDebugMessage(a, n, interfaceName, name);
      }
      return result ? result.interfaceName : 'any';
    };

```

**Comment by weidox** [comment]
@aperron Is this going to be fixed? Fix is provided in the duplicate issue, just need to change a line of code.

**Comment by aperron** [comment]
Sorry for the processing time.  
I just released [v2.0.7](https://github.com/erikvullings/strapi-to-typescript/releases/tag/v2.0.7) which should fix the component issues. Do not hesitate to give me a feedback.  

I missed your error message suggestions. But it's not lost, I keep it warm.

**Comment by jkerys-halo** [comment]
@aperron 2.0.7 does not work. I did a little debugging, in tsexporter.js there's a new piece of code which does properly parse component name, including prefix - which could help with an issue I wrote before where components with plural names were failing.
```
        this.strapiModels = strapiModelsParse.map((m) => {
            const modelName = m._isComponent ?
                path.dirname(m._filename).split(path.sep).pop() + '.' + path.basename(m._filename, '.json')
                : path.basename(m._filename, '.settings.json');
```
Here it parses `modelName` like `product.materials` or `product.amount` which is cool. However later there's something wrong because later I'm getting error `type 'material' unknown on Product[materials] => fallback to 'any'. Add in the input arguments the folder that contains *.settings.json with info.name === 'material'` and we're back to beginning..
